spec/controllers/contact/govuk/accessible_format_requests_controller_spec.rb

```diff
@@ -93,6 +93,62 @@
         expect(response.body).to have_css("form#request-accessible-format[action=\"/contact/govuk/request-accessible-format/sent\"]")
       end
     end
+
+    context "error messages" do
+      let(:required_inputs) { Hash[required_keys.collect { |v| [v.to_sym, example_param] }] }
+      context "questions with required inputs" do
+        let(:required_input_question) { format_request_questions.detect { |question| question["inputs"][0]["error_message"] } }
+        let(:required_input_question_number) { format_request_questions.find_index(required_input_question) + 1 }
+        describe "when a required input is not entered" do
+          before { post :form, params: base_params.merge({ 'question_number': required_input_question_number + 1, 'previous_question_number': required_input_question_number }) }
+          it "reposts a form requesting the previous question" do
+            expect(response.body).to have_css("input[name=question_number][value=#{required_input_question_number}]", visible: false)
+          end
+
+          it "sets the error messages in the flash" do
+            keys_and_messages = required_input_question["inputs"].collect do |input|
+              { input["key"] => input["error_message"] }
+            end
+
+            keys_and_messages.each do |key, message|
+              expect(flash["input_errors"][key]).to eq message
+            end
+          end
+        end
+
+        describe "when all required inputs are entered" do
+          let(:required_keys) { required_input_question["inputs"].filter_map { |input| input["key"] if input["error_message"] } }
+
+          before { post :form, params: required_inputs.merge(base_params.merge({ 'question_number': required_input_question_number + 1, 'previous_question_number': required_input_question_number })) }
+
+          it "proceeds to the next question" do
+            expect(response.body).to have_css("input[name=question_number][value=#{required_input_question_number + 1}]", visible: false)
+          end
+
+          it "does not set error messages in the flash" do
+            expect(flash["input_errors"]).to eq nil
+          end
+        end
+      end
+
+      context "questions with unrequired inputs" do
```

Comment:
Can we make this consistent with the naming in the earlier commits? I.e.

```suggestion
      context "questions with optional inputs" do
```

Reply:
Here I've tried to make a distinction between 
`optional questions` which may or may not appear depending on earlier questions vs
`required inputs` which are answers that must be provided. A question could have a mix of required (e.g email address) and non-required (e.g name) inputs

understand this naming may be confusing and/or complex. Perhaps `answers` rather than `inputs`?

Reply:
Oh ok. I think what you have or what I've suggested would both work in that case 👍 but perhaps add a comment above the `context` along the lines of what you've just explained, e.g. `# Questions can have a mix of required (e.g email address) and non-required (e.g name) inputs`